### PR TITLE
docs: add .github/copilot-instructions.md for Copilot agent context

### LINKED ISSUE #14: feat: add .github/copilot-instructions.md for Copilot agent context
## Problem

Copilot's automated PRs and review comments sometimes miss
project-specific conventions (e.g. `.dawon.toml` vs `.gato.toml`,
SHA-256 commitment model, no plaintext answers in source,
`#[cfg(unix)]` for POSIX-only tests).

GitHub supports a `.github/copilot-instructions.md` file that is
injected into every Copilot chat and code-review context for this
repository.

## Proposed solution

Add `.github/copilot-instructions.md` that mirrors `CLAUDE.md` in
content but is written in the Copilot custom-instructions format
(declarative rules, no shell commands).

Sections to include:

- **Project summary** — what Dawon does (stricter mini-moulinette,
  Rust, fork+pipe harness, SHA-256 commitments)
- **Config file** — always `.dawon.toml`, never `.gato.toml`
- **Harness design** — binary protocol, 65535-byte cap, no plaintext
  expected outputs anywhere in generated C
- **Platform guards** — `#[cfg(unix)]` for POSIX-only code/tests;
  `detect_leaks=1` is Linux-only (LSan not on macOS)
- **Commit discipline** — Conventional Commits 1.0.0, atomic commits,
  DCO Signed-off-by required
- **SHA-256 commitments** — verified with `printf 'expected' | sha256sum`
- **Adding subjects** — strict edge cases (INT_MIN, null byte, DEL 127)
- **No answer leakage** — never embed expected bytes in generated C

The file should mirror `CLAUDE.md` but in Copilot's declarative style.

## Priority

**Before v0.1.0** — so that Copilot reviews on open PRs (#4, #8, #9,
#11, #13) are context-aware.

## References

GitHub docs:
https://docs.github.com/en/copilot/customizing-copilot/adding-repository-custom-instructions-for-github-copilot

Existing conventions: `CLAUDE.md`, `CONTRIBUTING.md`

### PR DESCRIPTION
**Pull request by qlrd**
## Summary

Adds `.github/copilot-instructions.md` — the GitHub-supported file
that is injected into every Copilot chat, code-review, and agent
context for this repository.

Content mirrors `CLAUDE.md` but written in Copilot's declarative
instruction format (rules, not shell recipes):

- Config file is `.dawon.toml` (not `.gato.toml`)
- Harness is POSIX-only: `fork()`+`pipe()`, `#[cfg(unix)]` for tests
- SHA-256 commitments: no plaintext expected outputs in generated C
- `detect_leaks=1` in `ASAN_OPTIONS`: Linux only (LSan absent on macOS)
- Commit discipline: CC 1.0.0, atomic commits, DCO `Signed-off-by`
- Adding subjects: strict edge cases (`INT_MIN`, null byte, DEL 127)
- PR discipline: issue-first, rebase merge, all-commits-green rule

Closes #14.

## Test plan

- [ ] Read `.github/copilot-instructions.md` — all sections present
- [ ] Verify no `.gato.toml` references remain
- [ ] No Rust or CI changes; no `cargo test` run required

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: .github/copilot-instructions.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+- ASAN/UBSAN harness: fork+pipe binary protocol with SHA-256
+  commitment comparison in Rust
+- Valgrind: `--leak-check=full --error-exitcode=1` (Linux only)
+
+---
+
+## Configuration file
+
+The config file is always `.dawon.toml`.  Never write `.gato.toml`.
+
```
[the remark this answers is not in the document]
Comment (qlrd) [reply .github/copilot-instructions.md]: IMO this could even not be mentioned `.gato.toml` in every history of project. It was a human pun, @copilot

File: .github/copilot-instructions.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+## Platform guards
+
+| Feature | Guard |
+|---------|-------|
+| Round-trip harness tests | `#[cfg(unix)]` |
+| Valgrind step | `if: runner.os == 'Linux'` in CI |
+| norminette step | `if: runner.os != 'Windows'` in CI |
+| `detect_leaks=1` in ASAN_OPTIONS | Linux only — LSan is not available on macOS |
+
```
[the remark this answers is not in the document]
Comment (qlrd) [reply .github/copilot-instructions.md]: if macos cannot be targeted, isnt our problem, but wecomed for dev that wanna, @copilot . Add a good first issue on that.

[the remark this answers is not in the document]
Comment (qlrd) [reply .github/copilot-instructions.md]: The table in the latest commit is well-formed (single leading `|` on every row). If the malformation was in an earlier push it has been corrected. macOS is welcome for local development but is not in the CI matrix — a good-first-issue will be opened separately for contributors who want to add macOS CI support.

*— claude (@qlrd)*

File: .github/copilot-instructions.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+---
+
+## Platform guards
+
+| Feature | Guard |
+|---------|-------|
+| Round-trip harness tests | `#[cfg(unix)]` |
+| Valgrind step | `if: runner.os == 'Linux'` in CI |
+| norminette step | `if: runner.os != 'Windows'` in CI |
```
[the remark this answers is not in the document]
Comment (qlrd) [reply .github/copilot-instructions.md]: Fixed in the latest commit. CI is Linux-only; the multi-OS guard rows (`if: runner.os == ...`) have been moved under an explicit "if a multi-OS CI matrix is ever added" note so they are clearly advisory, not descriptive of the current state.

*— claude (@qlrd)*

**Comment by qlrd** [comment]
@copilot — CI is green, all review threads resolved. Ready to merge after PR #9.
